internal/virtualKubelet/node/api/server.go

```diff
@@ -47,8 +50,22 @@ func PodHandler(p PodHandlerConfig, debug bool) http.Handler {
 	if debug {
 		r.HandleFunc("/runningpods/", HandleRunningPods(p.GetPods)).Methods("GET")
 	}
+
+	r.HandleFunc("/pods", HandleRunningPods(p.GetPodsFromKubernetes)).Methods("GET")
 	r.HandleFunc("/containerLogs/{namespace}/{pod}/{container}", HandleContainerLogs(p.GetContainerLogs)).Methods("GET")
-	r.HandleFunc("/exec/{namespace}/{pod}/{container}", HandleContainerExec(p.RunInContainer)).Methods("POST")
+	r.HandleFunc(
+		"/exec/{namespace}/{pod}/{container}",
+		HandleContainerExec(
+			p.RunInContainer,
+		),
+	).Methods("POST", "GET")
+
+	if p.GetStatsSummary != nil {
```

Comment:
Why do we have this if? We are checking if the remote cluster has metrics enabled?

Reply:
This is for being sure the metrics handler has been set. Actually, this if is always true in our implementation, I can remove it.